ci/setup-and-build.ps1

```diff
@@ -0,0 +1,202 @@
+param(
+  [string] $gdk_home = (get-item "$($PSScriptRoot)").parent.FullName, ## The root of the UnrealGDK repo
+  [string] $gcs_publish_bucket = "io-internal-infra-unreal-artifacts-production"
+)
+
+$ErrorActionPreference = 'Stop'
+
+function Write-Log() {
+  param(
+    [string] $msg,
+    [Parameter(Mandatory=$false)] [bool] $expand = $false
+  )
+  if ($expand) {
+      Write-Output "+++ $($msg)"
+  } else {
+      Write-Output "--- $($msg)"
+  }
+}
+
+function Start-Event() {
+    param(
+        [string] $event_name,
+        [string] $event_parent
+    )
+
+    # Start this tracing span.
+    Start-Process -NoNewWindow "imp-ci" -ArgumentList @(`
+        "events", "new", `
+        "--name", "$($event_name)", `
+        "--child-of", "$($event_parent)"
+    ) | Out-Null
+
+    Write-Log "--- $($event_name)"
```

Comment:
Remove the leading `---` from here, since it's already pre-pended by the `Write-Log` function (thus this makes things log as `--- --- Unreal Engine` instead of the intended `--- Unreal Engine`.  

Sorry about that, I wrote this bug originally.  :)
```suggestion
    Write-Log "$($event_name)"
```